pkg/generator/caches.go

```diff
@@ -136,8 +141,20 @@ func (caches *Caches) ToEnvoySnapshot(ctx context.Context) (cache.Snapshot, erro
 	externalTLSVHosts := make([]*route.VirtualHost, 0, len(caches.translatedIngresses))
 	snis := sniMatches{}
 
+	localVHostsPerListener := make(map[string]portVHost)
+
 	for _, translatedIngress := range caches.translatedIngresses {
-		localVHosts = append(localVHosts, translatedIngress.internalVirtualHosts...)
+		if translatedIngress.listener != "" {
```

Comment:
I suspect we dont support going back and forth from `port` to `none` wrt isolation.
I am thinking of a scenario where existing ingress have this part set but the feature is not on.